test/integration/features/profile.feature

```diff
@@ -2,89 +2,162 @@
 Feature: Profile
   As a user I can perform basic operations of Minishift with profile feature
 
-  Scenario: Starting Minishift with default profile
-     Given Minishift has state "Does Not Exist"
-      When executing "minishift start" succeeds
-      Then Minishift should have state "Running"
-
-Scenario: User should be able to list default profile 'minishift'
-   Given Minishift has state "Running"
-    When executing "minishift profile list" succeeds
-    Then stdout should contain
-    """
-    - minishift	Running		(Active)
-    """
-  Scenario: Getting default profile internal docker registry address
-     Given Minishift has state "Running"
-      When executing "minishift openshift registry" succeeds
-      Then stdout should be valid IP with port number
+  Scenario: Starting Minishift with default profile 'minishift'
+    Given Minishift has state "Does Not Exist"
+     When executing "minishift start" succeeds
+     Then Minishift should have state "Running"
+
+  Scenario: User should be able to list default profile 'minishift'
+    Given Minishift has state "Running"
+     When executing "minishift profile list" succeeds
+     Then stdout should match "-\s*minishift\s*Running\s*\(Active\)"
+
+  Scenario: Getting default profile 'minishift' internal docker registry address
+    Given Minishift has state "Running"
+     When executing "minishift openshift registry" succeeds
+     Then stdout should be valid IP with port number
+
+  Scenario: User should be able to set a new profile 'foo'
+    Given Minishift has state "Running"
+     When executing "minishift profile set foo" succeeds
+     Then stdout should match "Profile\s*\'foo\'\s*set\s*as\s*active\s*profile"
+
+  Scenario Outline: User should be able to set and get values on profile 'foo'
+    Given Minishift has state "Does Not Exist"
```

Comment:
How do you get from "running state" in the previous scenario to "does not exist"? Does this test even pass?

Reply:
@hferentschik , I think due to `When executing "minishift profile set foo" succeeds` from previous scenario he switched to new profile where VM doesn't exist.

Reply:
@budhrg you are right.
@hferentschik To make the scenario more interactive we can add a good explanatory description to the scenario or can change the steps some thing like ```Profile foo has state "Does Not Exist"```

Reply:
@amitkrout +1 This really needs scenario description. Or I am even thinking about implementing and using a new step definition, which would make it explicit on which profile the status check is being executed, for example:

```gherkin
Given profile "foo" of Minishift has state "Running"
```
@budhrg @hferentschik WDYT?

Reply:
Done